### PR TITLE
feat(generate): default to js template

### LINKED ISSUE #3: put wasm behind a flag, default to js only worker
not actually sure we want this for 0.1.0

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #3 #71

### CODE REVIEW

File: src/commands/generate.rs [hunk 1]
```diff
@@ -36,3 +37,10 @@ fn command(name: &str, binary_path: PathBuf, args: &[&str]) -> Command {
     c.args(args);
     c
 }
+
+fn project_type(template: &str) -> ProjectType {
+    if template.contains("rust") {
```
Comment (steveklabnik) [inline src/commands/generate.rs]: This should probably be:

```suggestion
    if template == "rust" {
```

File: src/commands/generate.rs [hunk 2]
```diff
@@ -36,3 +37,10 @@ fn command(name: &str, binary_path: PathBuf, args: &[&str]) -> Command {
...
+        return ProjectType::Rust;
+    }
+    ProjectType::JavaScript
```
Comment (steveklabnik) [inline src/commands/generate.rs]: Also consider:

```rust
if template == "rust" {
    ProjectType::Rust
} else {
    ProjectType::JavaScript
}
```

Though this is of course subjective

File: src/commands/publish/mod.rs [hunk 3]
```diff
@@ -65,16 +39,48 @@ fn multi_script(user: &User, name: &str) -> Result<(), failure::Error> {
...
+    }
 
     Ok(())
 }
 
+fn build_js() -> Result<String, failure::Error> {
+    let mut f = fs::File::open("worker.js")?;
+    let mut buffer = String::new();
+    f.read_to_string(&mut buffer)?;
+    Ok(buffer)
```
Comment (steveklabnik) [inline src/commands/publish/mod.rs]: This function body can be replaced entirely with

```rust
std::fs::read_to_string("worker.js")
```

though, that returns an `io::Error`, so you *may* need

```rust
Ok(std::fs::read_to_string("worker.js")?)
```

File: src/main.rs [hunk 4]
```diff
@@ -130,11 +109,25 @@ fn main() -> Result<(), failure::Error> {
...
+            commands::build(&cache, &user.settings.project.project_type)?;
+        }
+
+        if let Some(matches) = matches.subcommand_matches("preview") {
+            let method = HTTPMethod::from_str(matches.value_of("method").unwrap_or("get"));
+
+            let body = match matches.value_of("body") {
+                Some(s) => Some(s.to_string()),
+                None => None,
+            };
```
Comment (steveklabnik) [inline src/main.rs]: This is

```rust
let body = matches.value_of("body").map(|s| s.to_string());
```

File: src/commands/generate.rs (hunk 1 above)
Comment (ashleygwilliams) [reply src/commands/generate.rs]: no- this is matching the url so contains is correct here

File: src/commands/generate.rs (hunk 2 above)
Comment (ashleygwilliams) [reply src/commands/generate.rs]: yeah i prefer it the way it currently is :)

File: src/commands/publish/mod.rs (hunk 3 above)
Comment (ashleygwilliams) [reply src/commands/publish/mod.rs]: yesss, as said in air i don't know why but the docs i end up getting on google search never show this very good optimized way

File: src/main.rs (hunk 4 above)
Comment (ashleygwilliams) [reply src/main.rs]: yeah i think i actually like this but friggin clippy thinks ur suggestion is a redundant closure and has no problem with what's there atm

**Pull request merged**